src/rofi_rbw/typer.py

```diff
@@ -75,3 +75,34 @@ def type_characters(self, characters: str, active_window: str) -> None:
             'wtype',
             characters
         ])
+
+class YDotoolTyper(Typer):
+    @staticmethod
+    def supported() -> bool:
+        return is_wayland() and is_installed('ydotool')
+
+    @staticmethod
+    def name() -> str:
+        return 'ydotool'
+
+    def get_active_window(self) -> str:
+        return "not possible with ydotool"
+
+    def type_characters(self, characters: str, active_window: str, use_root = False) -> None:
```

Comment:
I'm not sure what would be the best way to integrate `use_root` as a command-line argument, considering none of the other backends have extra options like this.

Reply:
That's indeed an interesting thing. I'm not on Wayland, so excuse my dumb questions 😉

Reply:
It does work without root if your user has access to `/dev/input/`, but ydotool also provides an option to run a daemon as root so that the delay workaround isn't needed. The sudo works if you have NOPASSWD set for ydotool, but I'm not sure if that's a good idea to include the root option at all considering the extra setup it needs (running daemon + suoders config).

Also, if you want to test, ydotool should work just fine with X11 as well, since it accesses the input devices directly. It even works in the tty and for things like ctrl+alt+f keys for console switching.

Reply:
I had a look at ydotool's documentation, but I'm still not sure about the setup: If you call `ydotool`, it tries to work with the daemon, and if that doesn't work, it uses `/dev/input` directly. Right?
So, is it enough if the daemon is started as `root` previously? If so, that would be my preferred solution.

Reply:
Do you (or anyone, really) have further input for this? Must `ydotool` be called with root?

Reply:
Sorry, I kinda abandoned this PR since at the time I was using gnome where this was needed, now I'm using sway where wtype works fine and is also much less of a headache to use.

`ydotool` must be called as root by default. If you change `/dev/input` to be writable by your user, then it can be used without root. You can run a daemon as root but it seems to only accept clients that also run as root for some reason? Could be a misconfiguration on my part. The main point of having a daemon is that `ydotool` registers a virtual input device, and it takes some time to be loaded by a compositor. The daemon manages the virtual device to avoid the delay/issues caused by creating an input device each time `ydotool` is called. I'm not 100% sure if the daemon allows unprivileged users to control it.

Technically the `ydotool` code in rofi-rbw is in a working state, but it needs to be documented how to use/set up `ydotool` itself. If any gnome/wayland users want to do so it would be welcome, but personally I don't need this anymore.

Reply:
Thanks for the update! :)

So I'll merge this code and we'll see if anyone needs more than this. However, I'll simplify the code a bit, as `use_root` is currently always false.